### ISSUE TITLE
Add more tests for set_index and sort_values in distributed

### ISSUE THREAD

**Issue by phofl** [dask/distributed#8170]
Recent bugs/regression we found made me look into our test coverage for both methods In distributed context. I think we should add basic tests for both methods to test the important configs

I am not looking for anything fancy here, just making sure that the keywords are covered. We barely have tests for these 2 methods in distributed, so that should be relatively low hanging fruit to improve things fur the future.

merge seems to be covered pretty well now.

These 2 issues made me look at this:

- https://github.com/dask/dask/pull/10493
- https://github.com/dask/distributed/pull/8121

cc @fjetter

**Comment by jrbourbeau** [comment]
We have lots of shuffling-related tests in over in `dask/dask`, unfortunately most (all?) aren't using p2p. Just a thought, we update the [`shuffle_method` fixture](https://github.com/dask/dask/blob/c488d49c4f9ebcf0e71918be1f258269b9a17c55/conftest.py#L95-L98) to include `p2p` to get more coverage

**Comment by fjetter** [comment]
Yeah, the `shuffle_method` fixture was actually introduced to enable paramterized tests for P2P. We never got that far and I believe there was some concern about test duration since the P2P tests would need to spin up a cluster
